apps/build-onchain-apps/src/hooks/useCollectionMetadata.ts

```diff
@@ -20,41 +26,81 @@ type CollectionMetadataResult =
       isLoading: false;
     };
 
+type JsonMetadata = {
+  name: string;
+  description: string;
+  image: string;
+};
+
+function tryParseMetadataJson(str: string): CollectionMetadataResult | undefined {
+  try {
+    const json = JSON.parse(str) as JsonMetadata;
+    return {
+      collectionName: json.name,
+      description: json.description,
+      imageAddress: ipfsToHTTP(json.image),
+      isLoading: false,
+    } as CollectionMetadataResult;
+  } catch (err) {}
+  return;
+}
+
 /**
  * @param enabled Whether the app is in a state where contracts can be queried.
+ * @param address Address for the contract
+ * @param abi ABI for the contract
+ * @param lookupType Lookup type to use for the contract URI function
+ * TODO: standardize once https://github.com/ethereum/ERCs/pull/150 is settled
  * @returns CollectionMetadataResult
  */
-function useCollectionMetadata(enabled: boolean) {
+function useCollectionMetadata(enabled: boolean, address: Address, abi: Abi) {
```

Comment:
This Hook is getting a bit too complex. I think for this PR is fine, but we need to do some refactoring and simplification as follow up.